### PR TITLE
Adding unknownDocument

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Updating the iOS Proto to match Android. This is in preparation for the held write acks removal.

### CODE REVIEW

File: Firestore/Protos/protos/firestore/local/maybe_document.proto [hunk 1]
```diff
@@ -20,14 +20,22 @@ message NoDocument {
   google.protobuf.Timestamp read_time = 2;
 }
 
-// Represents either an existing document or the explicitly known absence of a
-// document.
+// Represents either an existing document, the explicitly known absence of a document, or a document
```
Comment (rsgowman) [inline Firestore/Protos/protos/firestore/local/maybe_document.proto]: nit: Don't we line wrap to 80 chars for protos?

File: Firestore/Protos/protos/firestore/local/maybe_document.proto [hunk 2]
```diff
@@ -20,14 +20,22 @@ message NoDocument {
...
     NoDocument no_document = 1;
 
     // The document (if it exists).
     google.firestore.v1beta1.Document document = 2;
+
+    // Used if the document is known to exist but its data is unknown
+    UnknownDocument unknown_document = 3;
   }
+
+  // `hasCommittedMutations` marks documents that were written to the remote socument store based
```
Comment (rsgowman) [inline Firestore/Protos/protos/firestore/local/maybe_document.proto]: s/socument/document/

File: Firestore/Protos/protos/firestore/local/maybe_document.proto [hunk 3]
```diff
@@ -20,14 +20,22 @@ message NoDocument {
...
+  // on a write acknowledgment. These documents are potentially inconsistent with the  backend's
```
Comment (rsgowman) [inline Firestore/Protos/protos/firestore/local/maybe_document.proto]: nitnitnit: double space between 'the' and 'backend'

File: Firestore/Protos/protos/firestore/local/maybe_document.proto [hunk 4]
```diff
@@ -20,14 +20,22 @@ message NoDocument {
...
+  // copy and use the write's commit version as their document version.
+  bool hasCommittedMutations = 4;
```
Comment (rsgowman) [inline Firestore/Protos/protos/firestore/local/maybe_document.proto]: protos usually use snake_case for fields, i.e. 'has_committed_mutations'.

**Comment by rsgowman** [comment]
And while we're at it, this file is also missing a copyright header (as are the other files in this directory, though curiously not the protos in other directories.) Maybe fix that too? (Though maybe as a separate PR.)

(And apologies for the repeated post-review comments)

**Comment by schmidt-sebastian** [comment]
Copyright headers added as well

File: Firestore/Protos/protos/firestore/local/maybe_document.proto [hunk 5]
```diff
@@ -30,12 +56,13 @@ message MaybeDocument {
...
-    // Used if the document is known to exist but its data is unknown
+    // Used if the document is known to exist but its data is unknown.
     UnknownDocument unknown_document = 3;
   }
 
-  // `hasCommittedMutations` marks documents that were written to the remote socument store based
-  // on a write acknowledgment. These documents are potentially inconsistent with the  backend's
-  // copy and use the write's commit version as their document version.
-  bool hasCommittedMutations = 4;
+  // `hasCommittedMutations` marks documents that were written to the remote
```
Comment (rsgowman) [inline Firestore/Protos/protos/firestore/local/maybe_document.proto]: nit: consider fixing the name in the comment too.

File: Firestore/Protos/protos/firestore/local/target.proto [hunk 6]
```diff
@@ -22,6 +36,7 @@ message Target {
   // The last snapshot version received from the Watch Service for this target.
   //
   // This is the same value as TargetChange.read_time
+  // https://github.com/googleapis/googleapis/blob/master/google/firestore/v1beta1/firestore.proto#L734
```
Comment (rsgowman) [inline Firestore/Protos/protos/firestore/local/target.proto]: I don't much care, but this comment could go stale and end up pointing to the wrong thing if firestore.proto is changed in the future.

Solutions:
1) Nothing. If it changes, the reader will figure it out.
2) Embed the commit hash in the url, i.e. https://github.com/googleapis/googleapis/blob/02111201c4752cdd0d7a5714ec6644f1d0297f51/google/firestore/v1beta1/firestore.proto#L734
3) Omit "L734" and let the reader ctrl+f.

**Pull request merged**